apps/docs/content/guides/database/extensions/wrappers/overview.mdx

```diff
@@ -111,9 +111,84 @@ This process can be taxing on your database if you are moving large amounts of d
 
 ## Security
 
-Foreign Data Wrappers do not provide Row Level Security. We do not advise exposing them via your API. Wrappers should _always_ be stored in a private schema. For example, if you are connecting to your Stripe account, you should create a `stripe` schema to store all of your Foreign Tables inside. This schema should _not_ be added to the “Additional Schemas” setting in the API section.
-
-If you want to expose any of the columns to your public API, create a [View](https://supabase.com/docs/guides/database/tables#views) in the `public` schema. You can also interact with your Foreign Tables inside a [Database Function](https://supabase.com/docs/guides/database/functions).
+Foreign Data Wrappers do not provide Row Level Security. We do not advise exposing them via your API. Wrappers should _always_ be stored in a private schema. For example, if you are connecting to your Stripe account, you should create a `stripe` schema to store all of your foreign tables inside. This schema should _not_ be added to the “Additional Schemas” setting in the API section.
+
+If you want to expose any of the foreign table columns to your public API, you can create a [Database Function with security definer](https://supabase.com/docs/guides/database/functions#security-definer-vs-invoker) in the `public` schema, and then you can interact with your foreign tables through API.
+
+As an example, let's go to [SQL Editor](https://supabase.com/dashboard/project/_/sql/new) first and then follow below steps,
+
+1. Create a Stripe Products foreign table:
+
+   ```sql
+   create foreign table stripe.stripe_products (
+     id text,
+     name text,
+     active bool,
+     default_price text,
+     description text,
+     created timestamp,
+     updated timestamp,
+     attrs jsonb
+   )
+     server stripe_fdw_server
+     options (
+       object 'products',
+       rowid_column 'id'
+     );
+   ```
+
+2. Create a security definer function that queries the foreign table:
+
+   ```sql
+   create function public.get_stripe_products()
+   returns table (
+     id text,
+     name text,
+     active boolean,
+     default_price text,
+     description text
+   )
+   language plpgsql
+   security definer set search_path = public, extensions, pg_temp
+   as $$
+   begin
+     return query
+     select
+       t.id,
+       t.name,
+       t.active,
+       t.default_price,
+       t.description
+     from
+       stripe.stripe_products t;
```

Comment:
an example that contains a simple filter like filtering on `name` prefix would be a nice enhancement so users can see how to pass parameters through

Reply:
sure, add a function parameter in the sample code.